libs/hmpb/src/ballot_templates/nh_ballot_template.tsx

```diff
@@ -218,6 +243,374 @@ function BallotPageFrame({
+                    <WriteInLabel />
+                  </div>
+                </div>
+                <BubbleWrapper isWriteIn optionInfo={optionInfo} />
+              </li>
+            );
+          })}
+      </ul>
+    </Box>
+  );
+}
+
+function BallotMeasureContest({ contest }: { contest: YesNoContest }) {
+  return (
+    <Box style={{ padding: 0 }}>
+      <ContestHeader>
+        <DualLanguageText delimiter="/">
+          <h2>{electionStrings.contestTitle(contest)}</h2>
+        </DualLanguageText>
+      </ContestHeader>
+      <div
+        style={{
+          display: 'flex',
+          flexDirection: 'column',
+          justifyContent: 'space-between',
+          gap: '0.25rem',
+        }}
+      >
+        <div
+          style={{
+            padding: '0.5rem 0.5rem',
+            display: 'flex',
+            flexDirection: 'column',
+            gap: '0.5rem',
+          }}
+        >
+          <DualLanguageText>
+            <RichText
+              tableBorderWidth={'1px'}
+              tableBorderColor={Colors.DARKER_GRAY}
+              tableHeaderBackgroundColor={Colors.LIGHT_GRAY}
+            >
+              {electionStrings.contestDescription(contest)}
+            </RichText>
+          </DualLanguageText>
+        </div>
+        <ul
+          style={{
+            display: 'flex',
+            flexDirection: 'column',
+            justifyContent: 'end',
+          }}
+        >
+          {[contest.yesOption, contest.noOption].map((option) => (
+            <li
+              key={option.id}
+              style={{
+                padding: '0.375rem 0.5rem',
+                borderTop: `1px solid ${Colors.LIGHT_GRAY}`,
+              }}
+            >
+              <div
+                style={{
+                  display: 'flex',
+                  gap: '0.5rem',
+                  paddingTop: '0.25rem',
+                }}
+              >
+                <strong
+                  style={{ flex: 1, textAlign: 'right', marginTop: '-0.25rem' }}
+                >
+                  <DualLanguageText delimiter="/">
+                    {electionStrings.contestOptionLabel(option)}
+                  </DualLanguageText>
+                </strong>
+                <Bubble
+                  optionInfo={{
+                    type: 'option',
+                    contestId: contest.id,
+                    optionId: option.id,
+                  }}
+                />
+              </div>
+            </li>
+          ))}
+        </ul>
+      </div>
+    </Box>
+  );
+}
+
+function Contest({
+  contest,
+  election,
+}: {
+  contest: AnyContest;
+  election: Election;
+}) {
+  switch (contest.type) {
+    case 'candidate':
+      return <CandidateContest election={election} contest={contest} />;
+    case 'yesno':
+      return <BallotMeasureContest contest={contest} />;
+    default:
+      return throwIllegalValue(contest);
+  }
+}
+
+async function BallotPageContent(
```

Comment:
This is entirely copy + pasted from the default template. I considered adding a parameter to the default template's `BallotPageContent` to take a `Contest` component, which I may still do, but this was the fastest to unblock.

Reply:
> I considered adding a parameter to the default template's `BallotPageContent` to take a `Contest` component

I do like this idea! While I agree with the decision to keep the NH and Vx ballot templates separate instead of branching within a single template, I do think that we should keep duplication to a minimum where possible. I think there's a very high chance that a future fix in one template accidentally doesn't make it to the other template. Not blocking for the training of course but would be good to adjust while context is fresh.

Reply:
The approach I envisioned is that any component shared between templates goes into ballot_components.tsx to make it clear that they're shared.

I think figuring out the right abstractions will actually be a little tricky until we have a few templates and can see which parts differ, so I'm more ok with keeping copy-pasting as a starting point, even if it results in some duplication.

Reply:
I see. I'll open a follow up PR to move the components exported from `vx_default_ballot_template` to `ballot_components` while we decide how to handle `*Content` components.